internal/carver/cmd/main.go

```diff
@@ -0,0 +1,452 @@
+		return fmt.Errorf("unable to add replace module: %s", b)
+	}
+
+	cmd = exec.Command("go", "mod", "tidy")
+	cmd.Dir = rootMod.filePath
+	if b, err := cmd.Output(); err != nil {
+		return fmt.Errorf("unable to tidy parent module: %s", b)
+	}
+	return nil
+}
+
+func (c *carver) CreateGitTags(rootMod *modInfo) error {
+	futureTag, err := bumpSemverPatch(rootMod.tag)
+	if err != nil {
+		return err
+	}
+	childPrefix := strings.TrimPrefix(strings.TrimPrefix(c.childModPath, rootMod.filePath), "/")
+	log.Println("Commiting changes")
+	if !c.dryRun {
+		cmd := exec.Command("git", "add", "-A")
+		cmd.Dir = c.parentModPath
+		if b, err := cmd.Output(); err != nil {
+			return fmt.Errorf("unable to add changes: %s", b)
+		}
+		cmd = exec.Command("git", "commit", "-m",
+			fmt.Sprintf("chore(%s): carve out sub-module", childPrefix))
+		cmd.Dir = c.parentModPath
+		if b, err := cmd.Output(); err != nil {
+			return fmt.Errorf("unable to commit changes: %s", b)
+		}
+	}
+	log.Printf("Tagging Root module: %s", futureTag)
+	if !c.dryRun {
+		cmd := exec.Command("git", "tag", futureTag)
+		cmd.Dir = c.parentModPath
+		if b, err := cmd.Output(); err != nil {
+			return fmt.Errorf("unable to tag root module: %s", b)
+		}
+	}
+	log.Printf("Tagging Child module: %s/%s", childPrefix, c.childTagVersion)
+	if !c.dryRun {
+		cmd := exec.Command("git", "tag", fmt.Sprintf("%s/%s", childPrefix, c.childTagVersion))
+		cmd.Dir = c.parentModPath
+		if b, err := cmd.Output(); err != nil {
+			return fmt.Errorf("unable to tag child module: %s", b)
+		}
+	}
+	return nil
+}
+
+// newWriterCloser is wrapper for creating a file. Used for testing and
+// dry-runs.
+func (c *carver) newWriterCloser(fp string) (io.WriteCloser, error) {
+	if c.dryRun {
+		return noopCloser{w: io.Discard}, nil
+	}
+	if c.w != nil {
+		return noopCloser{w: c.w}, nil
+	}
+	return os.Create(fp)
+}
+
+// sortTags does a best effort sort based on semver. It was made a function for
+// testing. Only the top result will ever be used.
+func sortTags(tags []string) {
+	sort.Slice(tags, func(i, j int) bool {
+		imatch := semverRegex.FindStringSubmatch(tags[i])
+		jmatch := semverRegex.FindStringSubmatch(tags[j])
+		if len(imatch) < 5 {
+			return false
+		}
+		if len(jmatch) < 5 {
+			return true
+		}
+
+		// Matches must be numbers due to regex they are parsed from.
+		iM, _ := strconv.Atoi(imatch[1])
+		jM, _ := strconv.Atoi(jmatch[1])
+		im, _ := strconv.Atoi(imatch[2])
+		jm, _ := strconv.Atoi(jmatch[2])
+		ip, _ := strconv.Atoi(imatch[3])
+		jp, _ := strconv.Atoi(jmatch[3])
+
+		// weight each level of semver for comparison
+		iTotal := iM*marjorWeight + im*minorWeight + ip*patchWeight
+		jTotal := jM*marjorWeight + jm*minorWeight + jp*patchWeight
+
+		// de-rank all prereleases by a major version
+		if imatch[4] != "" {
+			iTotal -= marjorWeight
+		}
+		if jmatch[4] != "" {
+			jTotal -= marjorWeight
+		}
+
+		return iTotal > jTotal
+	})
+}
+
+func parsePkgName(importPath string) (string, error) {
```

Comment:
This is more like `guessPkgName` since the name may not always be the final element of the import path. Is there a better way?